docs/examples/301/log-analytics-with-datasources-solutions/main.bicep

```diff
@@ -0,0 +1,102 @@
+@description('Location of the workspace')
+param location string = resourceGroup().location
+@description('Name of the workspace')
+param name string
+@allowed([
+  'PerGB2018'
+  'Free'
+  'Standalone'
+  'PerNode'
+  'Standard'
+  'Premium'
+])
+@description('Sku of the workspace')
+param sku string
+@description('The workspace data retention in days, between 30 and 730')
+@minValue(7)
+@maxValue(730)
+param retentionInDays int
+@description('Solutions to add to workspace')
+param solutions array = []
+@description('Name of automation account to link to workspace')
+param automationAccountName string = ''
+@description('Datasources to add to workspace')
+param dataSources array = []
+@description('Enable lock to prevent accidental deletion')
+param enableDeleteLock bool = false
+@description('Enable diagnostic logs')
+param enableDiagnostics bool = false
+@description('Storage account name. Only required if enableDiagnostics is set to true.')
+param diagnosticStorageAccountName string = ''
+@description('Storage account resourceop. Only required if enableDiagnostics is set to true.')
+param diagnosticStorageAccountResourceGroup string = ''
+
+var lockName = '${logAnalyticsWorkspace.name}-lck'
+var diagnosticsName = '${logAnalyticsWorkspace.name}-dgs'
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' = {
+  name: name
+  location: location
+  properties: {
+    sku: {
+      name: sku
+    }
+    retentionInDays: retentionInDays
+  }
+}
+
+resource logAnalyticsSolutions 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = [for solution in solutions: {
+  name: '${solution.name}(${logAnalyticsWorkspace.name})'
+  location: location
+  properties: {
+    workspaceResourceId: logAnalyticsWorkspace.id
+  }
+  plan: {
+    name: '${solution.name}(${logAnalyticsWorkspace.name})'
+    product: solution.product
+    publisher: solution.publisher
+    promotionCode: solution.promotionCode
+  }
+}]
+
+resource logAnalyticsAutomation 'Microsoft.OperationalInsights/workspaces/linkedServices@2020-08-01' = if (!empty(automationAccountName)) {
+  name: '${logAnalyticsWorkspace.name}/Automation'

```

Comment:
Use either the `parent` property to declare this child resource or use the nested syntax. We are eventually going to deprecate the ability to declare child resources using this `/`-segmented name.

parent-style:

```bicep
resource logAnalyticsAutomation 'Microsoft.OperationalInsights/workspaces/linkedServices@2020-08-01' = if (...) {
  parent: logAnalyticsWorkspace
  
  name: 'Automation'
  ...
}
```

or

```bicep
resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' = {
  name: name
  ...

  resource logAnalyticsAutomation 'linkedServices' = {
    name: 'Automation'
    ...
   }
}
```

Reply:
done. changed to use parent (i prefer that over nested🙂)